### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by gayashanbc**

### CODE REVIEW

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java [hunk 1]
```diff
@@ -40,83 +42,94 @@
...
+
+        try {
+            return GrantHandlerServiceComponent.getRealmService().getTenantUserRealm(tenantId).getUserStoreManager();
+
+        } catch (UserStoreException e) {
+            throw new IdentityOAuth2Exception("Error occurred while getting user store manager", e);
+        }
+    }
+
+    private String ExtractUserFromSSHCertificate(String sshCertificate) {
```
Comment (omindu) [inline src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java:139]: Let's add a comment
```suggestion
        // The certificate will be in the format '<algorithm> <key> <comment>'
        // ex: 'ssh-ed25519-cert-v01@openssh.com AAAAB3Nza... user@host'.
        // We need to extract the 'key' component from the certificate
        String[] split = sshCertificate.trim().split("\\s+");
```

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java [hunk 2]
```diff
@@ -40,83 +42,94 @@
...
```
Comment (omindu) [inline src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]:
```suggestion
    private String extractUserFromSSHCertificate(String sshCertificate) {
```

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java [hunk 3]
```diff
@@ -40,83 +42,94 @@
...
-        if(subjectDN != null) {
-            username = subjectDN;
-            log.debug("Username is retrieved from Subject DN : " + username);
-        } else {
-            if (certParam != null) {
-                username = getUserFromSSHCert(certParam);
-            }
+        // Extract 'Principal' from the presented certificate.
+        if (certificateParameterValue != null) {
+            username = ExtractUserFromSSHCertificate(certificateParameterValue);
```
Comment (omindu) [inline src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]:
```suggestion
            username = extractUserFromSSHCertificate(certificateParameterValue);
```

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java [hunk 4]
```diff
@@ -40,83 +42,94 @@
...
+
+        } catch (UserStoreException e) {
+            throw new IdentityOAuth2Exception("Error occurred while getting user store manager", e);
+        }
+    }
+
+    private String ExtractUserFromSSHCertificate(String sshCertificate) {
 
         String[] split = sshCertificate.trim().split("\\s+");
         byte[] encodedCert = java.util.Base64.getMimeDecoder().decode(split[1].getBytes(UTF_8));
```
Comment (omindu) [inline src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]: let's check `split.length == 2`

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java [hunk 5]
```diff
@@ -40,83 +42,94 @@
...
+            return GrantHandlerServiceComponent.getRealmService().getTenantUserRealm(tenantId).getUserStoreManager();
+
+        } catch (UserStoreException e) {
+            throw new IdentityOAuth2Exception("Error occurred while getting user store manager", e);
+        }
+    }
+
+    private String ExtractUserFromSSHCertificate(String sshCertificate) {
 
         String[] split = sshCertificate.trim().split("\\s+");
```
Comment (omindu) [inline src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java:116]: Add a null check for `sshCertificate`

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java [hunk 6]
```diff
@@ -126,10 +139,11 @@ private String getUserFromSSHCert(String sshCertificate) {
 
                 if (publicKey instanceof Certificate) {
                     Certificate certificate = (Certificate) publicKey;
-
                     List<String> validPrincipals = certificate.getValidPrincipals();
+
                     if (validPrincipals != null && !validPrincipals.isEmpty()) {
                         String principal = validPrincipals.get(0);
```
Comment (omindu) [inline src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]:
```suggestion
                        // Only the 1st user principal is considered
                        String principal = validPrincipals.get(0);
```

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java [hunk 7]
```diff
@@ -40,83 +42,94 @@
...
-        String username = null;
-        if(subjectDN != null) {
-            username = subjectDN;
-            log.debug("Username is retrieved from Subject DN : " + username);
-        } else {
-            if (certParam != null) {
-                username = getUserFromSSHCert(certParam);
-            }
+        // Extract 'Principal' from the presented certificate.
+        if (certificateParameterValue != null) {
```
Comment (omindu) [inline src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]: Let's throw an error if `certificateParameterValue` is `null`

Comment (gayashanbc) [reply src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java (hunk 6 above)
Comment (gayashanbc) [reply src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java (hunk 4 above)
Comment (gayashanbc) [reply src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java (hunk 3 above)
Comment (gayashanbc) [reply src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java (hunk 2 above)
Comment (gayashanbc) [reply src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1

File: src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java (hunk 1 above)
Comment (gayashanbc) [reply src/main/java/org/wso2/sample/identity/oauth2/grant/ed25519/X509GrantHandler.java:139]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/3/commits/d01c3a2846a13850ced1a2e93b47bb46687eb8e1

**Review by omindu — APPROVED** [review]

**Pull request merged**